docs/developer/designs/min-runtime/README.md

```diff
@@ -0,0 +1,118 @@
+# [DRAFT] Minimum runtime before preemptions and reclaims
+
+## Overview
+
+This document proposes a new feature called "reclaim-min-runtime" and "preempt-min-runtime" that provides configurable guarantees for job runtime before preemption or reclaim can occur. This feature enables administrators to define minimum runtime guarantees at various levels of the resource hierarchy: node pool, nth level queue and leaf-queue.
+
+## Motivation
+
+Unrestrained preemption can lead to resource thrashing, where jobs are repeatedly preempted before making meaningful progress. The min-runtime feature addresses this issue by providing configurable minimum runtime guarantees that can be set by either cluster operators or sometimes within parts of the queue to provide guarantees about minimal useful work.
+
+## Detailed Design
+
+### Reclaim Min Runtime Configuration
+
+The reclaim-min-runtime parameter can be configured with the following values:
+
+- **0 (default)**: Jobs are always preemptible via reclaims
+- **-1**: Jobs are never preemptible via reclaims
+- **Positive value (in seconds)**: Minimum guaranteed runtime before preemption via reclaims
```

Comment:
I don't think it's necessary to specify that we'll use seconds here, IMO we should use a format accepted by time.ParseDuration (https://www.geeksforgeeks.org/time-parseduration-function-in-golang-with-examples/), like we do for scheduling period or staleness-grace-period (see https://github.com/NVIDIA/KAI-Scheduler/blob/main/cmd/scheduler/app/options/options.go), but either way - no need to commit to anything at this point